include/LightGBM/config.h

```diff
@@ -140,6 +140,8 @@ struct TreeConfig: public ConfigBase {
   int num_leaves = 127;
   int feature_fraction_seed = 2;
   double feature_fraction = 1.0;
+  // max cache size(unit:GB) for historical histogram. < 0 means not limit
```

Comment:
yes, but considering this option will be used only on machine with limited memories, offer an    fine-grained option will be nice. you can multiply 1024 to make code unchanged